logstash-core/spec/logstash/instrument/namespaced_metric_spec.rb

```diff
@@ -27,6 +27,90 @@
     new_namespace = subject.namespace(:wally)
 
     expect(subject.namespace_name).to eq([namespace])
-    expect(new_namespace.namespace_name).to eq([:stats, :wally])
+    expect(new_namespace.namespace_name).to eq([:root, :wally])
+  end
+
+  context "#increment" do
+    it "a counter by 1" do
+      metric = subject.increment(:error_rate)
+      expect(collector).to be_a_metric_event([:root, :error_rate], :counter, :increment, 1)
+    end
+
+    it "a counter by a provided value" do
+      metric = subject.increment(:error_rate, 20)
+      expect(collector).to be_a_metric_event([:root, :error_rate], :counter, :increment, 20)
+    end
+
+    it "raises an exception if the key is an empty string" do
```

Comment:
Not strictly required, but consider adding the `raises an exception if the key is an empty string` and `raise an exception if the key is nil` specs to a shared example like "crud operations". these are repeated in increment/decrement/gauge and, though it is not tested, probably also affect `#time`.